packages/components/checkbox/tests/use-checkbox.test.tsx

```diff
@@ -0,0 +1,267 @@
+      },
+    )
+
+    expect(result.current.checked).toBeFalsy()
+
+    rerender({ checked: true })
+
+    expect(result.current.checked).toBeTruthy()
+  })
+
+  test("should have icon props", () => {
+    const { result } = renderHook(() => useCheckbox({}))
+
+    const { getIconProps } = result.current
+    expect(typeof getIconProps).toBe("function")
+
+    const props = getIconProps({}, null)
+
+    expect(props).toHaveProperty("onMouseDown")
+    expect(typeof props.onMouseDown).toBe("function")
+
+    expect(props).toHaveProperty("onMouseEnter")
+    expect(typeof props.onMouseEnter).toBe("function")
+
+    expect(props).toHaveProperty("onMouseLeave")
+    expect(typeof props.onMouseLeave).toBe("function")
+
+    expect(props).toHaveProperty("onMouseUp")
+    expect(typeof props.onMouseUp).toBe("function")
+  })
+
+  test("should have container props", () => {
+    const { result } = renderHook(() => useCheckbox({}))
+
+    const { getContainerProps } = result.current
+    expect(typeof getContainerProps).toBe("function")
+
+    const props = getContainerProps({}, null)
+
+    expect(props).toHaveProperty("onClick")
+    expect(typeof props.onClick).toBe("function")
+  })
+
+  test("should have label props", () => {
+    const { result } = renderHook(() => useCheckbox({}))
+
+    const { getLabelProps } = result.current
+    expect(typeof getLabelProps).toBe("function")
+
+    const props = getLabelProps({}, null)
+
+    expect(props).toHaveProperty("onMouseDown")
+    expect(typeof props.onMouseDown).toBe("function")
+
+    expect(props).toHaveProperty("onTouchStart")
+    expect(typeof props.onTouchStart).toBe("function")
+  })
+
+  test("should have input props", () => {
+    const { result } = renderHook(() => useCheckbox({}))
+
+    const { getInputProps } = result.current
+    expect(typeof getInputProps).toBe("function")
+
+    const props = getInputProps({}, null)
+
+    expect(props).toHaveProperty("onBlur")
+    expect(typeof props.onBlur).toBe("function")
+
+    expect(props).toHaveProperty("onChange")
+    expect(typeof props.onChange).toBe("function")
+
+    expect(props).toHaveProperty("onFocus")
+    expect(typeof props.onFocus).toBe("function")
+
+    expect(props).toHaveProperty("onKeyDown")
+    expect(typeof props.onKeyDown).toBe("function")
+
+    expect(props).toHaveProperty("onKeyUp")
+    expect(typeof props.onKeyUp).toBe("function")
+
+    expect(props).toHaveProperty("style")
+  })
+
+  test("should call default onMouseDown on label props when no handler is passed", () => {
+    const InputBox = () => {
+      const { getLabelProps } = useCheckbox({})
+      const labelProps = { ...getLabelProps() }
+
+      return <input type="checkbox" data-testid="input-box" {...labelProps} />
+    }
+
+    render(<InputBox />)
+    const InputBoxComponent = screen.getByTestId("input-box")
+    fireEvent.mouseDown(InputBoxComponent)
+    expect(InputBoxComponent).toBeInTheDocument()
+  })
+
```

Comment:
Tests after this are not very meaningful.
This is because even if the execution of a `fireEvent` fails, it cannot be detected.
It would be good to create a component using `useCheckbox` and see if it works somewhat correctly.

Reply:
Rewrote these tests. Now we are using a custom component to test use-checkbox hook.